includes/data_functions.js

```diff
@@ -11,7 +11,7 @@ function eventDataExtract(dataField, keyToExtract, dynamic = false) {
   return `NULLIF(
     (
       SELECT
-        ARRAY_TO_STRING(ARRAY_CONCAT_AGG(value), ",") as concat_value
+        IF(ARRAY_LENGTH(ARRAY_CONCAT_AGG(value)) = 0, NULL, ARRAY_TO_STRING(ARRAY_CONCAT_AGG(value), ",")) AS concat_value
```

Comment:
Now that we're setting columns to NULL when the source columns are NULL, will you need a migration to fix existing data in BQ ?

Reply:
@asatwal good point. Although I don't think a migration would be possible because we have no way to tell whether a past empty string actually *was* an empty string, or whether it was a NULL. So if it matters to teams they'll need to backfill.

Also to check we are setting columns to ```value:[]``` now, not ```value:[NULL]``` right? The latter will cause BQ errors while I think the former is fine.

Reply:
@stevenleggdfe - For non-array columns even before the fix to the DfE Analytics GEM, NULL values are `value:[]` and empty columns are `value:['']`. It's only for array columns that the fix to the GEM removes the NULL value.

Reply:
Great :-)